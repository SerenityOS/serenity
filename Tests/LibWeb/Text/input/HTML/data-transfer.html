<script src="../include.js"></script>
<script>
    asyncTest(async done => {
        let dataTransfer = new DataTransfer();
        println(`dropEffect: ${dataTransfer.dropEffect}`);
        println(`effectAllowed: ${dataTransfer.effectAllowed}`);

        let dataTransferItemList = dataTransfer.items;
        println(`length=${dataTransferItemList.length}, types=${dataTransfer.types}`);

        let stringItem = dataTransferItemList.add("well hello friends", "custom-type");
        println(`stringItem: kind=${stringItem.kind}, type=${stringItem.type}`);
        println(`length=${dataTransferItemList.length}, types=${dataTransfer.types}`);

        if (dataTransferItemList[0] !== stringItem) {
            println("FAILED");
        }

        let promise = new Promise((resolve, reject) => {
            stringItem.getAsString(data => {
                println(`stringItemAsString: data=${data}`);
                resolve();
            });
        });
        await promise;

        try {
            dataTransferItemList.add("well hello friends", "custom-type");
            println("FAILED");
        } catch (e) {}

        let file = new File(["well hello friends"], "file.txt", {
            type: "text/plain",
        });

        let fileItem = dataTransferItemList.add(file);
        println(`fileItem: kind=${fileItem.kind}, type=${fileItem.type}`);
        println(`length=${dataTransferItemList.length}, types=${dataTransfer.types}`);

        let fileItemAsFile = fileItem.getAsFile();
        println(`fileItemAsFile: name=${fileItemAsFile.name}, type=${fileItemAsFile.type}`);

        let fileItemAsEntry = fileItem.webkitGetAsEntry();
        println(`fileItemAsEntry: name=${fileItemAsEntry.name}, file=${fileItemAsEntry.isFile}, directory=${fileItemAsEntry.isDirectory}`);

        if (dataTransferItemList[1] !== fileItem) {
            println("FAILED");
        }

        if (dataTransferItemList[2] !== undefined) {
            println("FAILED");
        }

        done();
    });
</script>
