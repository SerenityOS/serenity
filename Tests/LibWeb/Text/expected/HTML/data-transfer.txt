dropEffect: none
effectAllowed: none
length=0, types=
stringItem: kind=string, type=custom-type
length=1, types=custom-type
stringItemAsString: data=well hello friends
fileItem: kind=file, type=text/plain
length=2, types=custom-type,Files
fileItemAsFile: name=file.txt, type=text/plain
fileItemAsEntry: name=file.txt, file=true, directory=false
